src/DIRAC/Core/Tornado/Client/private/TornadoBaseClient.py

```diff
@@ -505,6 +507,12 @@ def _request(self, retry=0, outputFile=None, **kwargs):
     # Do we use the server certificate ?
     if self.kwargs[self.KW_USE_CERTIFICATES]:
       cert = Locations.getHostCertificateAndKeyLocation()
+    elif self.kwargs.get(self.KW_PROXY_STRING):
+      tmpHandle, cert = tempfile.mkstemp()
```

Comment:
Does this ever get cleaned up again?

Reply:
I think you can make `cert` a tuple to avoid writing anything to disk?

Reply:
No it does not, and I don't really like it, but in fact we never clean them up... I don't see an esthetic way of cleaning it without ugly flags and `if`.

I did hope being able to pass certs as strings, but it is not supported unfortunately

https://github.com/psf/requests/blob/10968357a19a146332c8dfd8e6d17006d746e580/requests/adapters.py#L242-L254

Reply:
I don't think the file needs to exist outside of this function's scope?

If so I think you can used `NamedTemporaryFile` and let it delete it as part of the garbage collection sometime after the function exits?

Reply:
I thought of using `NamedTemporaryFile`, but
* There is that slightly bothering statement in the doc (https://docs.python.org/3/library/tempfile.html#tempfile.NamedTemporaryFile) 
> Whether the name can be used to open the file a second time, while the named temporary file is still open, varies across platforms (it can be so used on Unix; it cannot on Windows NT or later).
* I still need to close the file eventually. Where do I close it ?

Reply:
You don't need to close it, you can write then flush and the the GC take care of closing it. It's an ugly hack that doesn't work on Windows but it's better than leaving the files lying around and we're already very broken on the latter.

Reply:
Damn you ! Your last comment lost me in `cpython` implementation and posix specification for a while :D 

> let Python delete it as part of the garbage collection sometime after the function exits

There is actually no guarantee that the file will ever be closed. The `inputoutput` tutorial used to say so up to 3.7, but was then changed to reflect the reality:

https://docs.python.org/3.7/tutorial/inputoutput.html#reading-and-writing-files
> If you’re not using the with keyword, then you should call f.close() to close the file and immediately free up any system resources used by it. If you don’t explicitly close a file, Python’s garbage collector will eventually destroy the object and close the open file for you, but the file may stay open for a while. Another risk is that different Python implementations will do this clean-up at different times.

https://docs.python.org/3.8/tutorial/inputoutput.html#reading-and-writing-files
> If you’re not using the with keyword, then you should call f.close() to close the file and immediately free up any system resources used by it.
Warning
Calling f.write() without using the with keyword or calling f.close() might result in the arguments of f.write() not being completely written to the disk, even if the program exits successfully. 

Full discussion about the matter:
https://bugs.python.org/issue17852
https://github.com/python/cpython/pull/23528/files

And since `NamedTemporaryFile` relies on close being called (https://github.com/python/cpython/blob/064464fc38269e70f7e3a34cb25fc9085ab85782/Lib/tempfile.py#L501-L505)....

Now, I can add a `finally` in the big `try/except` and do some cleanup there, based on ugly flag. I agree with you that leaving the file behind sucks, but the use case is so seldom, it can only be performed by someone with a lot of karma already, and finally, it is safe from the file permission point of view.